File: backend/app/schemas/user_notify_channel.py
"""
用户通知渠道 Pydantic Schema
NOTIFY-CORE 实现
"""

from datetime import datetime
from typing import Any, Optional
from pydantic import BaseModel, Field

from app.models.enums.user_notify_channel_type import UserNotifyChannelType


class UserNotifyChannelBase(BaseModel):
    """用户通知渠道基础模型"""
    channel_type: UserNotifyChannelType
    display_name: Optional[str] = None
    config: dict[str, Any] = Field(default_factory=dict)
    is_enabled: bool = True


class UserNotifyChannelCreate(UserNotifyChannelBase):
    """创建用户通知渠道"""
    pass


class UserNotifyChannelUpdate(BaseModel):
    """更新用户通知渠道"""
    display_name: Optional[str] = None
    config: Optional[dict[str, Any]] = None
    is_enabled: Optional[bool] = None


class UserNotifyChannelRead(UserNotifyChannelBase):
    """读取用户通知渠道"""
    id: int
    is_verified: bool
    last_test_at: Optional[datetime] = None
    last_test_ok: Optional[bool] = None
    last_error: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class UserNotifyChannelTestResponse(BaseModel):
    """测试通知响应"""
    success: bool
    message: str

File: frontend/src/types/userNotifyChannel.ts
/**
 * 用户通知渠道类型定义
 * NOTIFY-CORE 实现
 */

export type UserNotifyChannelType = 'telegram_bot' | 'webhook' | 'bark'

export interface UserNotifyChannel {
  id: number
  channel_type: UserNotifyChannelType
  display_name?: string | null
  config: Record<string, any>
  is_enabled: boolean
  is_verified: boolean
  last_test_at?: string | null
  last_test_ok?: boolean | null
  last_error?: string | null
  created_at: string
  updated_at: string
}

export interface UserNotifyChannelCreate {
  channel_type: UserNotifyChannelType
  display_name?: string | null
  config?: Record<string, any>
  is_enabled?: boolean
}

export interface UserNotifyChannelUpdate {
  display_name?: string | null
  config?: Record<string, any>
  is_enabled?: boolean
}

export interface UserNotifyChannelTestResponse {
  success: boolean
  message: string
}

// 渠道类型选项
export const channelTypeOptions = [
  { value: 'telegram_bot', title: 'Telegram 机器人', icon: 'mdi-telegram', description: '通过 Telegram Bot 接收通知' },
  { value: 'webhook', title: 'Webhook', icon: 'mdi-webhook', description: '发送 HTTP 请求到自定义 URL' },
  { value: 'bark', title: 'Bark', icon: 'mdi-bell-ring', description: 'iOS 推送通知' },
] as const

// 渠道配置字段定义
export const channelConfigFields: Record<UserNotifyChannelType, { key: string; label: string; type: string; required?: boolean; placeholder?: string }[]> = {
  telegram_bot: [
    // Telegram 通过绑定流程配置，不需要手动输入
  ],
  webhook: [
    { key: 'url', label: 'Webhook URL', type: 'url', required: true, placeholder: 'https://example.com/notify' },
    { key: 'secret', label: 'Secret（可选）', type: 'password', placeholder: '用于验证请求' },
  ],
  bark: [
    { key: 'server', label: '服务器地址', type: 'url', required: true, placeholder: 'https://api.day.app/your-key' },
    { key: 'sound', label: '提示音', type: 'text', placeholder: 'alarm' },
    { key: 'group', label: '分组', type: 'text', placeholder: 'VabHub' },
  ],
}
